src/ui/public/query_bar/components/typeahead/suggestion.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { EuiIcon } from '@elastic/eui';
+import classNames from 'classnames';
+
+function getEuiIconType(type) {
+  switch (type) {
+    case 'field':
+      return 'kqlField';
+    case 'value':
+      return 'kqlValue';
+    case 'recentSearch':
+      return 'search';
+    case 'conjunction':
+      return 'kqlSelector';
+    case 'operator':
+      return 'kqlOperand';
+    default:
+      throw new Error('Unknown type', type);
+  }
+}
+
+export function Suggestion(props) {
```

Comment:
Could I make a suggestion? (Pardon the pun). Here's how I've seen a lot of components formatted both in EUI and in Kibana:

```jsx
export function Suggestion({
  selected,
  suggestion,
  onMouseEnter,
  onClick,
  innerRef,
  ariaId,
}) {
  const {
    text,
    description,
    type,
  } = suggestion;

  const classes = classNames('typeahead-item', { active: selected });
  const itemClasses = `suggestionItem suggestionItem--${type}`;

  return (
    <div
      className={classes}
      role="option"
      onClick={() => onClick(suggestion)}
      onMouseEnter={onMouseEnter}
      ref={innerRef}
      id={ariaId}
    >
      <div className={itemClasses}>
        <div className="suggestionItem__type" type={type}>
          <EuiIcon type={getEuiIconType(type)}/>
        </div>
        <div className="suggestionItem__text">{text}</div>
        <div
          className="suggestionItem__description"
          // Description currently always comes from us and we escape any potential user input
          // at the time we're generating the description text
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: description }}
        />
      </div>
    </div>
  );
}
```

The notable differences are the use of destructuring and the extraction of calculated values out of the JSX itself. At first I didn't like the use of destructuring but it grew on me because it helped me identify which props I was using up-front, and it also reduced noise in the JSX. Just a suggestion, feel free to disregard.

Reply:
I'm mostly cool with destructuring and happy to follow whatever conventions folks are using for react stuff already, will make the update 👍